artemis-server/src/main/java/org/apache/activemq/artemis/core/management/impl/view/ConsumerView.java

```diff
@@ -82,7 +82,7 @@ public Object getField(ServerConsumer consumer, String fieldName) {
          case "user":
             return session.getUsername();
          case "clientID":
-            return consumer.getConnectionClientID();
+            return session.getMetaData("jms-client-id") == null ? consumer.getConnectionClientID() : session.getMetaData("jms-client-id");
```

Comment:
this should not be needed, if anything the session and consumer models should be re-worked to allow for a client id to be set at session level generically.